starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,430 @@
 // Find the maximum
+function maxOfTwoNumbers(a, b) {
+  if (a > b) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
 // Finding Longest Word
+
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
 
+function findLongestWord(words) {
+  //console.log(words);
+  var longest = "";
+
+  for (var i = 0; i < words.length; i++) {
+    if (words[i].length > longest.length) {
+      longest = words[i];
+    } else if (words.length === 1) {
+      console.log(words[i]);
+    }
+  }
+  console.log("The longest word is :" + longest + " .");
+}
+
+findLongestWord(words);
+
 // Calculating a Sum
 
-var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumArray(numbers) {
+  // var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+  var reducer = (accumulator, currentValue) => accumulator + currentValue;
+
+  if (numbers.length < 1) {
+    return 0;
+  } else {
+    return numbers.reduce(reducer);
+  }
+}
+
+sumArray(numbers);
 
 // Calculate the Average
 
-var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers(numbers) {
+  //  var numbers = [2, 6, 9, 10, 7, 4, 1, 9];
+  var reducer = (accumulator, currentValue) => accumulator + currentValue;
+  if (numbers.length < 1) {
+    return undefined;
+  } else {
+    var average = numbers.reduce(reducer) / numbers.length;
+    return average;
+  }
+}
+averageNumbers(numbers);
 
 // Array of Strings
 var wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(wordsArr) {
+  var sum = "";
+  var num = wordsArr.length;
+  console.log(num);
+  console.log(sum);
+  for (var i = 0; i < num; i++) {
+    sum += num;
+
+    return sum / num;
+  }
+}
+
```

Comment:
Your function can't work because you're not passing the variable in into the `for loop`.
You need to pass all the elements of the array and add their length to the sum. 

You could have done it like this: 
```
function averageWordLength(array) {
  if (array.length === 0) {
    return undefined;
  }
  var sumLetters = 0;
  for (var i = 0; i < array.length; i++) {
    sumLetters += array[i].length;
  }
  return sumLetters / array.length;
}
```